src/frost.rs

```diff
@@ -284,22 +286,37 @@ fn generate_shares<R: RngCore + CryptoRng>(
         commitment.0.push(Commitment(SpendAuth::basepoint() * c));
     }
 
+    // Participants' identifiers are generated as follows:
+    // id = H("FROST_id", g^s, i), where:
+    // g is the basepoint,
+    // s is the secret chosen by the dealer, and
+    // i is the index of the participant.
+    // The hash is finalized in the loop that follows.
+    let mut hasher = HStar::default();
+    hasher
+        .update("FROST_id".as_bytes())
+        .update(jubjub::AffinePoint::from(SpendAuth::basepoint() * secret.0).to_bytes());
+
```

Comment:
We should break this out into a function that takes the secret as an argument.

Reply:
This description indicates to me that we should change it to be a separate hasher state per id, because:

> id_i = H(g^s || i || "some domain separator") for ids 1...n

is what we want, not

> id_i = H(g^s || i_1 || ... || i_n || "some domain separator") for id n

which is basically what we have now.

Reply:
Yes, agreed with @dconnolly here